examples/browser/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+extern crate midir;
+extern crate console_error_panic_hook;
+extern crate js_sys;
+extern crate web_sys;
+extern crate wasm_bindgen;
+
+use js_sys::{Array};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsCast;
+use web_sys::{console};
+
+use std::sync::{Arc, Mutex};
+use std::error::Error;
+
+use midir::{MidiInput, Ignore};
+
+pub fn log(s: String) {
+    console::log(&Array::of1(&s.into()));
+}
+
+macro_rules! println {
+    ()              => (log("".to_owned()));
+    ($($arg:tt)*)   => (log(format!($($arg)*)));
+}
+
+#[wasm_bindgen(start)]
+pub fn start() {
+    std::panic::set_hook(Box::new(console_error_panic_hook::hook));
+
+    let token_outer = Arc::new(Mutex::new(None));
+    let token = token_outer.clone();
+    let closure : Closure<dyn FnMut()> = Closure::wrap(Box::new(move ||{
+        if run().unwrap() == true {
+            if let Some(token) = *token.lock().unwrap() {
+                web_sys::window().unwrap().clear_interval_with_handle(token);
+            }
+        }
+    }));
+    *token_outer.lock().unwrap() = web_sys::window().unwrap().set_interval_with_callback_and_timeout_and_arguments_0(
+        closure.as_ref().unchecked_ref(),
+        200,
+    ).ok();
+    closure.forget();
+}
+
+fn run() -> Result<bool, Box<dyn Error>> {
+    let mut midi_in = MidiInput::new("midir reading input")?;
+    midi_in.ignore(Ignore::None);
+
+    // Get an input port (read from console if multiple are available)
+    let in_port = match midi_in.port_count() {
+        0 => return Ok(false),
```

Comment:
Why do you return `Ok(false)` here without printing a message "no input port found"? This is confusing, because when I run the test (as you described) without any input device, simply nothing happens (except that Chrome asks for permission to use MIDI).

Reply:
I originally removed it because it's possibly a bit spammy before the user first permits MIDI access, but that's probably better than being confusing so I've added it back 👍

Reply:
Then the message should explain that no ports will be found until you grant access to the browser.